doc/man7/migration_guide.pod

```diff
@@ -0,0 +1,1311 @@
+was constructed using this key in the first place, for example using a function
+or macro such as EVP_PKEY_assign_RSA(), EVP_PKEY_set1_RSA(), etc.
+Where the EVP_PKEY holds a provider managed key, then these functions now return
+a cached copy of the key. Changes to the internal provider key that take place
+after the first time the cached key is accessed will not be reflected back in
+the cached copy. Similarly any changes made to the cached copy by application
+code will not be reflected back in the internal provider key.
+
+For the above reasons the keys returned from these functions should typically be
+treated as read-only. To emphasise this the value returned from
+EVP_PKEY_get0_RSA(), EVP_PKEY_get0_DSA(), EVP_PKEY_get0_EC_KEY() and
+EVP_PKEY_get0_DH() has been made const. This may break some existing code.
+Applications broken by this change should be modified. The preferred solution is
+to refactor the code to avoid the use of these deprecated functions. Failing
+this the code should be modified to use a const pointer instead.
+The EVP_PKEY_get1_RSA(), EVP_PKEY_get1_DSA(), EVP_PKEY_get1_EC_KEY()
+and EVP_PKEY_get1_DH() functions continue to return a non-const pointer to
+enable them to be "freed". However they should also be treated as read-only.
+
+=head4 Deprecated i2d and d2i functions for low level key types.
+Any i2d and d2i functions such as d2i_DHparams() that take a low level key type
+have been deprecated. Applications should instead use the OSSL_DECODER and
+OSSL_ENCODER APIs to read and write files.
+See L<d2i_RSAPrivateKey(3)/Migration> for further details.
+
+=Head4 Deprecated low level key parameter getters
+
+Functions that access low level objects directly such as RSA_get0_n() are now
+deprecated. Applications should use one of L<EVP_PKEY_get_bn_param(3)>,
+L<EVP_PKEY_get_int_param(3)>, l<EVP_PKEY_get_size_t_param(3)>,
+L<EVP_PKEY_get_utf8_string_param(3)>, L<EVP_PKEY_get_octet_string_param(3)> or 
+L<EVP_PKEY_get_params(3)> to access fields from an EVP_PKEY.
+Gettable parameters are listed in L<EVP_PKEY-RSA(7)/Common RSA parameters>,
+L<EVP_PKEY-DH(7)/DH parameters>, L<EVP_PKEY-DSA(7)/DSA parameters>,
+L<EVP_PKEY-FFC(7)/FFC parameters>, L<EVP_PKEY-EC(7)/Common EC parameters> and
+L<EVP_PKEY-X25519(7)/Common X25519, X448, ED25519 and ED448 parameters>.
+Applications may also use EVP_PKEY_todata() to return all fields.
+
+=Head4 Deprecated low level key parameter setters
+
+Functions that access low level objects directly such as RSA_set0_crt_params()
+are now deprecated. Applications should use L<EVP_PKEY_fromdata(3)> to create
+new keys from user provided key data. Keys should be immutable once they are
+created so if required the user may use EVP_PKEY_todata(), OSSL_PARAM_merge(),
+and EVP_PKEY_fromdata() to create a modified key.
+See L<EVP_PKEY-DH(7)/Examples> for more information.
+See L<Deprecated low level key generation functions> for information on generating
+a key using parameters.
+
+EVP_PKEY_copy_parameters
```

Comment:
Was there supposed to be some text here?

Reply:
No I just plonked that there to remind that I need to insert it elsewhere.

Reply:
removed this.